.github/workflows/standard_spa_pr.yml

```diff
@@ -0,0 +1,463 @@
+        if: ${{ !env.HAS_SKIP }}
+
+  chromatic-deployment:
+    defaults:
+      run:
+        working-directory: ${{ inputs.working_directory }}
+    runs-on: ${{ inputs.use_github_runner && inputs.github_runner || fromJson(inputs.default_runner) }}
+    timeout-minutes: 15
+    concurrency:
+      group: chromatic-deployment-${{ needs.create-shared-vars.outputs.groupId }}
+      cancel-in-progress: true
+    needs: [create-shared-vars]
+    if: ${{ inputs.use_chromatic }}
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 18.15.0
+          cache: 'npm'
+          registry-url: 'https://registry.npmjs.org'
+        env:
+          NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
+      - run: npm ci
+
+      - name: Publish to Chromatic
+        uses: chromaui/action@v1
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          projectToken: ${{ secrets.CHROMATIC_PROJECT_TOKEN }}
+          onlyChanged: true
+          exitOnceUploaded: true
+
+  deploy_magic_url:
+    defaults:
+      run:
+        working-directory: ${{ inputs.working_directory }}
+    name: Deploy To Dev
+    runs-on: ${{ inputs.use_github_runner && inputs.github_runner || fromJson(inputs.default_runner) }}
+    timeout-minutes: 15
+    concurrency:
+      group: deploy-magic-url-${{ needs.create-shared-vars.outputs.groupId }}
+      cancel-in-progress: true
+    needs: [create-shared-vars]
+    if: ${{ inputs.use_magic_url }}
+    permissions:
+      id-token: write
+      contents: read
+      pull-requests: write
+    outputs:
+      migration: ${{ steps.packageJson.outputs.migration }}
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 18.15.0
+          cache: 'npm'
+          registry-url: 'https://registry.npmjs.org'
+        env:
+          NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
+      - run: npm ci
+      - id: packageJson
+        name: Get Package Migration Number
+        run: |
+          echo "migration=$(jq -r .config.migration < package.json)" >> $GITHUB_OUTPUT
+      - name: Building
+        run: npm run build
+      - name: configure aws credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          role-to-assume: ${{ secrets.AWS_ROLE }}
+          role-session-name:
+            ${{ github.event.repository.name }}-magic-url-role-session
+          aws-region: ${{ secrets.AWS_REGION }}
+      - name: Deploy
+        # This bucket file location is static and editing it will break the Magic URL. This pushes the entire directory which includes the bundle and remote types if applicable
+        run: |
+          aws s3 sync deploy/dist s3://${{ secrets.AWS_APPS_BUCKET }}/static/manual-deploy/${{ github.event.repository.name }}@${{ steps.packageJson.outputs.migration }}/PR-${{ github.event.number }}/
```

Comment:
Push the contents of the build to an S3 bucket that now contains a reference to both the name of the application along with the migration number.